torch/_functorch/_aot_autograd/traced_function_transforms.py

```diff
@@ -792,21 +792,35 @@ def create_functional_call(mod, params_spec, params_len, store_orig_mod=False):
     # https://github.com/pytorch/pytorch/issues/103569
 
     def functional_call(*args, **kwargs):
+        nonlocal mod
+        # Need to unwrap and go down the fx.Interpreter.run() path below
+        # in order to properly preserve node stacktraces
+        mod_ = mod.gm if isinstance(mod, torch._dynamo.utils.GmWrapper) else mod
+
         with stateless._reparametrize_module(
-            mod, pytree.tree_unflatten(args[:params_len], params_spec)
+            mod_, pytree.tree_unflatten(args[:params_len], params_spec)
         ), maybe_disable_thunkify():
-            if isinstance(mod, torch.fx.GraphModule):
+            if isinstance(mod, (torch.fx.GraphModule, torch._dynamo.utils.GmWrapper)):
                 with fx_traceback.preserve_node_meta(), warnings.catch_warnings():
                     warnings.filterwarnings(
                         "ignore", "Anomaly Detection has been enabled."
                     )
                     with torch.autograd.detect_anomaly(check_nan=False):
                         detect_fake_mode().epoch += 1
-                        out = PropagateUnbackedSymInts(mod).run(
-                            *args[params_len:], **kwargs
-                        )
+                        if isinstance(mod, torch._dynamo.utils.GmWrapper):
+                            # Assumption : GmWrapper is only used by compiled autograd.
+                            # it stores no buffers/params and has no kwargs.
+                            # we need to carefully **not** unwrap the args list
+                            # to ensure args are boxed.
+                            assert params_len == 0
+                            assert len(kwargs) == 0
+                            out = PropagateUnbackedSymInts(mod_).run(args)
```

Comment:
there's some logic in GmWrapper.forward that we'll need here: https://github.com/pytorch/pytorch/blob/90d2593b3e819002157cab90d92351fcfe3ac59b/torch/_dynamo/utils.py#L2907-L2909

Reply:
You should probably just have a "middleware" wrapper that uniformly takes care of unwrapping GmWrapper and modifying the calling convention, should be cleaner.

@xmfan is there a reason we HAVE to have a GmWrapper? Shouldn't custom GraphModule prelude/postlude be enough here?

Reply:
What is the prelude/postlude? We use GmWrapper to work around the dynamo GraphModule needing boxed inputs, but AOTDispatcher always tracing the GraphModule with flat inputs

Reply:
Code generation for fx.Graph can be overridden via _codegen field. For example, this is used to generate GraphModule that can take arbitrary pytree as argument, it manages flattening/unflattening in the body. You could potentially use a similar mechanism to implement GmWrapper. cc @suo @Chillee

Reply:
I can look into it, but we probably still need GmWrapper for non-dynamo frontends that are passing in non-overriden graphs